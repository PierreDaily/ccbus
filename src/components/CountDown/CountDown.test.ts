import { mount, shallowMount } from "@vue/test-utils";
import CounDown from "./index.vue";

const timeOut = "2 min";

test("Mount component", async () => {
  // arrange
  expect(CounDown).toBeTruthy();


  const wrapper = mount(CounDown, {
    props: {
        timeLeft:timeOut
    },
  });

  // assert

  expect(wrapper.html()).toMatchSnapshot();
});

test("Should render '2 min' when time props is passed on", async () => {
  // arrange
  expect(CounDown).toBeTruthy();

  const wrapper = mount(CounDown, {
    props: {
        timeLeft: timeOut
    },
  });

  // assert
  expect(wrapper.text()).toContain(timeOut);

});


test("Should render '----' when time props isn't passed on", async () => {
    // arrange
    expect(CounDown).toBeTruthy();
  
    const wrapper = mount(CounDown, {
      props: {
      },
    });
  
    // assert
    expect(wrapper.text()).toContain("---");
  
  });
