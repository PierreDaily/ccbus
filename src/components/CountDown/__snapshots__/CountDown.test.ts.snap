// Vitest Snapshot v1

exports[`Mount component 1`] = `"<div data-v-44ef50d8=\\"\\"><img aria-label=\\"coming in\\" src=\\"/src/assets/icons/rss.svg\\" data-v-44ef50d8=\\"\\" /><span style=\\"width: 100px;\\" data-v-44ef50d8=\\"\\">2 min</span></div>"`;
